grails-app/services/au/org/ala/biocollect/merit/DocumentService.groovy

```diff
@@ -100,4 +102,65 @@ class DocumentService {
 
         documents;
     }
+
+    /**
+     * Returns true if the current user has permission to edit/update the
+     * supplied document.
+     * @param document the document to be edited/updated
+     */
+    boolean canEdit(Map document) {
+        if (document.documentId) {
+            document = get(document.documentId)
+        }
+        boolean canEdit = false
+
+        if (document) {
+            // Only FC_ADMINS can edit an existing read only document, but
+            // other roles can create them.
+            if (document.readOnly && document.documentId) {
+                canEdit = userService.userIsAlaOrFcAdmin()
+            }
+            else {
+                canEdit = hasEditorPermission(document)
+            }
+
+        }
+
+        canEdit
+    }
+
+    /** Returns true if the currently logged in user has permission to edit the supplied Document */
+    private boolean hasEditorPermission(Map document) {
+        boolean canEdit = userService.userIsAlaOrFcAdmin()
+        if (!canEdit) {
+            // Check the permissions that apply to the entity the document is
+            // associated with.
+            String userId = userService.getCurrentUserId()
+            if (document.projectId) {
+                canEdit = userService.canUserEditProject(userId, document.projectId)
+            } else if (document.organisationId) {
+                canEdit = userService.isUserAdminForOrganisation(userId, document.organisationId)
+            }
+            else if (document.activityId) {
+                Map activity = activityService.get(document.activityId)
```

Comment:
I propose that since this is an improvement to the current situation I can make documents with an activityId publicly viewable for now and it can be addressed when the other document changes are made.

Reply:
that sounds good.